### PR TITLE
Universal Profiling: add probabilistic profiling

### PR DESCRIPTION
**Pull request by florianl**
We have implemented probabilistic profiling in https://github.com/elastic/prodfiler/issues/3162. This PR adds the end user facing documentation so it can be configured if required.
 
Fixes https://github.com/elastic/prodfiler/issues/3163

### CODE REVIEW

File: docs/en/observability/profiling-probabilistic-profiling.asciidoc [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+The option `-probabilistic-interval` expects a time duration and defines the time interval for
+which probabilistic profiling will be enabled or disabled. Its default value is 1 minute.
+
+The following example shows how to configure the Universal Profiling Agent with a threshold of 50
+and an interval of 2 minutes and 30 seconds:
+
+[source,bash]
+----
+sudo pf-host-agent/pf-host-agent -probabilistic-threshold=50 -probabilistic-interval=2m30s ...'
+----
```
Comment (christos68k) [inline docs/en/observability/profiling-probabilistic-profiling.asciidoc:22]: I would add a real-world example here at the end, describing what this configuration amounts to in practical terms. For instance, something like:

"Over time and across the agent fleet, this configuration is expected to result in  ~50% of deployed agents collecting profiles at any given time. This reduces storage costs by accepting the trade-off of not having profile collection enabled all the time on all agents."

**Review by christos68k — APPROVED** [review]
LGTM

File: docs/en/observability/profiling-probabilistic-profiling.asciidoc (hunk 1 above)
Comment (florianl) [reply docs/en/observability/profiling-probabilistic-profiling.asciidoc:22]: added suggestion.

**Comment by florianl** [comment]
@mdbirnstiehl the CI pipeline failes with the following error:
```
asciidoctor: WARNING: invalid reference: profiling-probabilistic-profiling
```

Do you have an idea how this can be resolved?

**Comment by mdbirnstiehl** [comment]
> @mdbirnstiehl the CI pipeline failes with the following error:
> 
> ```
> asciidoctor: WARNING: invalid reference: profiling-probabilistic-profiling
> ```
> 
> Do you have an idea how this can be resolved?

I'll take a look and give you a full review today. We will need to add the new page to docs/en/observability/index.asciidoc so it shows up in the navigation. This could be the issue potentially.

File: docs/en/observability/profiling-advanced-configuration.asciidoc [hunk 2]
```diff
@@ -8,4 +8,5 @@ See the following sections for more information:
 
 * <<profiling-tag-data-query, Tag data for querying>>: Tag data collected by the host-agent into multiple logical groups so they can be queried in Kibana.
 * <<profiling-add-symbols, Add symbols for native frames>>: Push symbols to your cluster so you can see function names and line numbers in traces of applications written in programming languages that compile to native code (C, C++, Rust, Go, etc.).
-* <<profiling-use-a-proxy,Use a proxy>>:  Set up an HTTP proxy if your infrastructure host-agent installation needs one to reach {ecloud}.
\ No newline at end of file
+* <<profiling-use-a-proxy,Use a proxy>>:  Set up an HTTP proxy if your infrastructure host-agent installation needs one to reach {ecloud}.
+* <<profiling-probabilistic-profiling, Configure probabilistic profiling>>: Configure Universal Profiling Agent to run in probabilistic profiling mode.
```
Comment (mdbirnstiehl) [inline docs/en/observability/profiling-advanced-configuration.asciidoc:12]:
```suggestion
* <<profiling-probabilistic-profiling, Configure probabilistic profiling>>: Configure your host-agent to run in probabilistic profiling mode.

```

File: docs/en/observability/profiling-probabilistic-profiling.asciidoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+Configuring probabilistic profiling is a viable option to reduce storage costs.
+
+The option `-probabilistic-threshold` expects an unsigned integer as argument. If set to a value
+between 1 and 99 will enable probabilistic profiling. Every probabilistic interval a random number
+between 0 and 99 is chosen. If the given probabilistic threshold is greater than this random
+number, the agent will collect profiles from this system for the duration of the interval.
+
+The option `-probabilistic-interval` expects a time duration and defines the time interval for
+which probabilistic profiling will be enabled or disabled. Its default value is 1 minute.
+
```
Comment (mdbirnstiehl) [inline docs/en/observability/profiling-probabilistic-profiling.asciidoc:15]:
```suggestion

[[discrete]]
== Example

```

File: docs/en/observability/profiling-probabilistic-profiling.asciidoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
+[[profiling-probabilistic-profiling]]
+= Configure probabilistic profiling
+
+beta::[]
+
+Configuring probabilistic profiling is a viable option to reduce storage costs.
+
```
Comment (mdbirnstiehl) [inline docs/en/observability/profiling-probabilistic-profiling.asciidoc:7]:
```suggestion

[[discrete]]
== Configure probabilistic profiling

To configure probabilistic profiling,  set the `-probabilistic-threshold` and `probabilistic-interval` options.

```

File: docs/en/observability/profiling-probabilistic-profiling.asciidoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+[source,bash]
+----
+sudo pf-host-agent/pf-host-agent -probabilistic-threshold=50 -probabilistic-interval=2m30s ...'
+----
+
+Over time and across the Universal Profiling Agent fleet, this configuration is expected to result
+in ~50% of deployed agents collecting profiles at any given time. This reduces storage costs by
+accepting the trade-off of not having profile collection enabled all the time on all Universal
+Profiling Agents.
```
Comment (mdbirnstiehl) [inline docs/en/observability/profiling-probabilistic-profiling.asciidoc:27]:
```suggestion

```

**Review by mdbirnstiehl — COMMENTED** [review]
I recommended a few changes. Let me know if you have any questions or comments. 

If you add:

`include::profiling-probabilistic-profiling.asciidoc[leveloffset=+3]`

after 

`include::profiling-use-a-proxy.asciidoc[leveloffset=+3]`

in the `// Universal Profiling` section of `observability-docs\en\observability\index.asciidoc` file, you should pass the CI.

File: docs/en/observability/profiling-probabilistic-profiling.asciidoc [hunk 6]
```diff
@@ -0,0 +1,30 @@
+[[profiling-probabilistic-profiling]]
+= Configure probabilistic profiling
+
+beta::[]
+
+Probabilistic profiling allows you to reduce storage costs by collecting a representative sample of profiling data. 
+Over time and across the Universal Profiling agent fleet, this configuration results in approximately 50% of deployed agents collecting profiles at any given time. 
+This reduces storage costs with the trade-off that profile collection is not enabled at all times on all Universal Profiling agents.
+
+[[discrete]]
```
Comment (mdbirnstiehl) [inline docs/en/observability/profiling-probabilistic-profiling.asciidoc:10]:
```suggestion
[discrete]
```

**Review by mdbirnstiehl — APPROVED** [review]
Sorry about that, I formatted the discrete tags incorrectly in my review. If you fix those, that should fix the CI this time. After that, LGTM!

File: docs/en/observability/profiling-advanced-configuration.asciidoc (hunk 2 above)
Comment (florianl) [reply docs/en/observability/profiling-advanced-configuration.asciidoc:12]: replaced Universal Profiling Agent with Profiling Host Agent as suggested by @iogbole

**Pull request merged**